packages/treasure/src/Treasure.ts

```diff
@@ -2,12 +2,31 @@ import {
   type TransactionFilter,
   type SwapActionParams,
   compressJson,
+  type MintActionParams,
 } from '@rabbitholegg/questdk'
 import { type Address } from 'viem'
 import { MAGICSWAP_TOKENS, V2_ROUTER } from './constants'
-import { V2_ROUTER_ABI } from './abi'
+import { MINT_TREASURE_TAG_ABI, V2_ROUTER_ABI } from './abi'
 import { Chains, buildV2PathQuery } from './utils'
 
+export const mint = async (
+  mint: MintActionParams,
+): Promise<TransactionFilter> => {
+  const { chainId, contractAddress, recipient } = mint
+
+  return compressJson({
+    chainId,
+    to: contractAddress,
+    from: recipient,
+    input: {
+      $abi: MINT_TREASURE_TAG_ABI,
+      _registerArgs: {
+        owner: recipient,
```

Comment:
is this `owner` param redundant given that it will always be the same as the `from` param?

Reply:
Yeah, either/or is fine